Core/Barcelona/IMCoreHelpers/HookManager.swift

```diff
@@ -103,37 +103,90 @@ private func IDSQueryHooks() throws -> Interpose {
                 ) -> Void>
             ) in
             { controller, message, queue, replyBlock, failBlock, waitForReply  in
+                let logFailureBlock: (NSError?) -> Void = { [failBlock] error in
+                    log.error("Got failure for __sendMessage, error is \(String(describing: error))")
+                    failBlock(error)
+                }
+
                 let logReplyBlock: (OS_xpc_object) -> Void = { [replyBlock] replyObject in
+                    Task {
```

Comment:
I think this might be cleaner with full `try`/`catch`, would avoid the nested `do` blocks somewhat, and reduce the logging in the `guard`s. No need to fix now, just an idea for when you are refactoring.